components/Login/index.js

```diff
@@ -0,0 +1,115 @@
+import { signIn, signOut, useSession } from "next-auth/react";
+import styled from "styled-components";
+import Image from "next/image";
+import { useState } from "react";
+
+export default function Login({ isHomepage }) {
+  const { data: session } = useSession();
+  const [isHovered, setIsHovered] = useState(false);
+
+  function handleSignIn() {
+    signIn("github", { callbackUrl: "/exercises" });
+  }
+
+  const ButtonComponent = isHomepage ? HomepageButton : AnimatedButton;
+
+  return (
+    <ButtonComponent
+      onClick={session ? () => signOut() : handleSignIn}
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
+      onTouchStart={() => setIsHovered(!isHovered)}
+      $isHovered={isHovered}
+    >
+      {session ? (
+        isHomepage ? (
+          "Log Out"
+        ) : (
+          <>
```

Comment:
That's right: Since it is wrapped inside the Button component you don't need the fragments here.